app/controllers/vendor_sweets_controller.rb

```diff
@@ -0,0 +1,18 @@
+class VendorSweetsController < ApplicationController
+  def create
+    @vendor = Vendor.find(vendor_sweet_params[:vendor_id])
+    @vendor_sweet = @vendor.vendor_sweets.build(vendor_sweet_params)
```

Comment:
Although this works, you don't need to use `build` here since `vendor_id` is already included in the params.